src/execution.rs

```diff
@@ -96,6 +96,15 @@ impl CompletedTask {
     }
 }
 
+/// After a task finishes execution, it produces a result to be returned to
+/// raclette's caller.
+#[derive(Debug)]
+pub struct TaskResult {
+    pub full_name: Vec<String>,
+    pub duration: Duration,
+    pub status: Status,
+}
+
```

Comment:
Looks very similar to `CompletedTask`. Maybe use it instead? It looks more useful as it also contains stdout/stderr.

Reply:
Isn't the "reporting" consuming the `CompletedTask`? I first thought of only returning a boolean, then decided to return the status and finally settled for name, status and duration.

Reply:
Done